clang/lib/Driver/Driver.cpp

```diff
@@ -2809,6 +2809,12 @@ bool Driver::checkForOffloadStaticLib(Compilation &C,
       !Args.hasArg(options::OPT_fopenmp_targets_EQ))
     return false;
 
+  /// This fix will not work when mixing Xilinx devices with devices that use
+  /// OffloadStaticLibs
+  for (auto &P : llvm::make_range(C.getOffloadToolChains<Action::OFK_SYCL>()))
+    if (P.second->getTriple().isXilinxFPGA())
+      return false;
+
   // Right off the bat, assume the presense of -foffload-static-lib means
```

Comment:
```suggestion
  // Right off the bat, assume the presence of -foffload-static-lib means
```

Reply:
This code is from intel so i don't think modifying it even a typo id good idea because it increases likelihood of conflicts.

Reply:
OK. I made https://github.com/intel/llvm/pull/3079 for this